static-checks/rpmbuild-ctest/test.py

```diff
@@ -27,14 +27,42 @@
     ]
     util.subprocess_run(cmd, cwd=build_dir)
 
-    with open(build_dir / 'ctest_results') as f:
-        # Result format: X/Y Test  #X: test_name .................  test_result   Z sec
-        result_regex = re.compile(r'\d+\s+Test\s+#\d+:\s+([^\s]+)\s+\.+')
+    with open(build_dir / 'Testing' / 'Temporary' / 'LastTest.log') as f:
+        state = 'start'
+        test_output = []
+
         for line in f:
-            result_match = result_regex.search(line)
-            if result_match:
-                test_name = result_match.group(1)
-                result = 'pass' if 'Passed' in line else 'fail'
-                results.report(result, test_name)
+            line = line.rstrip('\n')
+
+            if state == 'output':
+                test_output.append(line)
+                if '<end of output>' in line:
+                    state = 'after_output'
+                continue
+
+            if state == 'after_output' and re.match(r'^-{10,}\s*$', line):
+                test_output.append(line)
+                state = 'result'
+                continue
+
+            if (
+                state == 'start'
+                and (m := re.fullmatch(r'^[0-9]+/[0-9]+\s+Testing:\s+(.+)$', line))
+            ):
```

Comment:
This is valid and you can keep it like this, but consider splitting it into two conditions to make it (IMHO) look better:
```suggestion
            if state == 'start':
                if m := re.fullmatch(r'^[0-9]+/[0-9]+\s+Testing:\s+(.+)$', line):
```
This is not possible every time, but here you do `continue` anyway, which breaks up the flow, so it can be done here.